### PR TITLE
Optimize index and line range search

### PR DESCRIPTION
**Pull request by tdroxler**
Improve the places where we were using `.split("\n|\r\n|\r")`, we now loop over each char and stop when we found what we need.

### CODE REVIEW

**Review by simerplaha — COMMENTED** [review]
👍🏼

Very minor things.

File: compiler-access/src/main/scala/org/alephium/ralph/lsp/access/compiler/message/SourceIndexExtra.scala [hunk 1]
```diff
@@ -53,30 +53,83 @@ object SourceIndexExtra {
...
+    }
+  }
 
-      val separatorLength = lineSeparatorLength(code)
-      val lineNumberLookup =
-        codeLines(code)
-         .foldLeft(Seq(0))((acc, line) => acc :+ (acc.last + line.size + separatorLength))
+  /**
+   * Constructs a line range from the provided source code, using the specified start and end indexes.
+   * The while loop optimizes performance by stopping once the necessary indexes are found.
```
Comment (simerplaha) [inline compiler-access/src/main/scala/org/alephium/ralph/lsp/access/compiler/message/SourceIndexExtra.scala:64]: 🚀

File: compiler-access/src/test/scala/org/alephium/ralph/lsp/access/compiler/message/SourceIndexExtraSpec.scala [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+              from = LinePosition(startLineIndex, startLine.indexOf(">>")),
+              to = LinePosition(endLineIndex, endLine.replaceFirst(">>", "").indexOf("<<") - 1)
+            )
+        }.head
+
+    val codeWithoutSymbol =
+      code.replaceAll(">>|<<", "")
+
+    buildLineRange(codeWithoutSymbol,from, to) shouldBe expectedLineRange
+  }
```
[the remark this answers is not in the document]
Comment (tdroxler) [reply compiler-access/src/test/scala/org/alephium/ralph/lsp/access/compiler/message/SourceIndexExtraSpec.scala:34]: yes, but I will need to extract that function in `compiler-access`

[the remark this answers is not in the document]
Comment (simerplaha) [reply compiler-access/src/test/scala/org/alephium/ralph/lsp/access/compiler/message/SourceIndexExtraSpec.scala:34]: Oh yep. Or this test-case could be pulled into `presentation-compiler`?

**Comment by tdroxler** [comment]
@simerplaha I did a bunch of new small commit to clean/move/extracts things

**Review by simerplaha — APPROVED** [review]
🚀

**Pull request merged**